### DISCUSSION TITLE
getStaticProps revalidate does not take string int

### DISCUSSION THREAD

**Discussion by khenrix** [vercel/next.js#19709] [Ideas]
# Feature request
Make revalidate param accept integer strings.

## Is your feature request related to a problem? Please describe.
I have a dev/prod environment for one of my applications and wanted to have a different revalidate time for dev / prod. I added an environment variable and made a revalidate in getStaticProps read from it. I however noticed that I needed to parse the environment variable to an Integer as "revalidate" was not able to read an int in string format. I felt like that should be possible and could cause confusion for other developers as well.  

## Describe the solution you'd like

When a string is passed as a revalidate input. Try parsing as int and accept if valid int, reject otherwise.

## Describe alternatives you've considered
- Better documentation and describe the default values in the docs. 
- Confuse more developers and let it be as is

**Comment by balazsorban44** [comment]
I can back this, could be a nice change.


Although not something that can't be handled already easily. We define revalidate in our .env files and in pipelines. For it to work though, I currently write this in `getStaticProps`:

`revalidate: Number(process.env.REVALIDATE)`

**Comment by khenrix** [comment]
Yes, I agree, it's something that can be managed quite easily now.
We used "parseInt(process.env.OUR_REVALIDATE_ENV_VAR,10)" instead.  

But maybe we should at least clarify the type in the docs.
`revalidate - An optional amount in seconds after which a page re-generation can occur. More on Incremental Static Regeneration`

They are clear enough to eventually understand that "oh, this says seconds, maybe it needs to be of type number" after some debugging.
